### ISSUE TITLE
How to use GPU batch size > 1?

### ISSUE THREAD

**Issue by ethanjperez** [rrmenon10/ADAPET#6]
What is the correct way to use a larger GPU batch size than 1 during training? Should I ​increase the `batch_size` in the config file (e.g., [here](https://github.com/rrmenon10/ADAPET/blob/master/config/BoolQ.json#L5)) by a factor of N, while decreasing `grad_accumulation_factor` by a factor of N?

I'm asking because I'm not sure if the code supports GPU batch size greater than 1, e.g., [this comment](https://github.com/rrmenon10/ADAPET/blob/master/src/adapet.py#L245) says a function assumes batch size 0 (?).

Any pointers here (or other tips for speeding up training) would be greatly appreciated :) Thanks for your help!

**Comment by rrmenon10** [comment]
The method you mention to increase the batch sizes is correct. However, we have only implemented that option for single-label tasks in this repo. 

For multi-label tasks (ReCoRD, COPA, WSC), the computations didn’t quite allow us to accommodate more than 1 (batch size). 

For evaluation and testing on ReCoRD (which takes the most time), we found it helpful to split the eval data and parallelize the computation (using the pre trained model) for eval labels before merging them together and computing the accuracy.
